### PR TITLE
feat(curriculum) : Added Node and Express Quiz

### LINKED ISSUE #56237: [HACKTOBERFEST 2024] - Node and Express Quiz
> [!IMPORTANT]
> This issue is part of our 2024 Hacktoberfest initiative. If you are interested in working on this issue, refer to our [top-level tasks](https://github.com/freeCodeCamp/freeCodeCamp/issues/56170) to see if this issue is claimed, or to claim it if it is not.

This issue is to create the Node and Express quiz.

The file you should be working on is found in `/curriculum/challenges/english/25-front-end-development/quiz-node-and-express`.

**All quiz questions need to relate to one of these topics:**

- JavaScript runtimes and engines (V8, node, deno, bun, browsers)
- npm
- Express
- connect node to SQL db

> [!CAUTION]
> Questions which do not fall under these topics will be rejected.

You are welcome to include code blocks in your questions, but please keep them short (four lines or less).

Please do NOT open a pull request until you have completed all questions. Your quiz should have **exactly 20 questions**.

### PR DESCRIPTION
**Pull request by nitesh2920**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #56237

### CODE REVIEW

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md [hunk 1]
```diff
@@ -17,375 +17,374 @@ Answer all of the questions below correctly to pass the quiz.
 
 #### --text--
 
-Placeholder question
+Which command initializes a new package.json file in npm?
 
 #### --distractors--
 
-Placeholder distractor 1
+npm create
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:24]:
```suggestion
`npm create`
```
you need to use backticks when writing code

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md [hunk 2]
```diff
@@ -17,375 +17,374 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+To enable Cross-Origin Resource Sharing
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which SQL command is used to delete records from a database?
```
[the remark this answers is not in the document]
Comment (nitesh2920) [reply curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:240]: okay any more updation required ?

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md [hunk 3]
```diff
@@ -17,375 +17,374 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+SpiderMonkey
 
 ---
 
-Placeholder distractor 3
+Chromium
```
Comment (lasjorg) [inline curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:76]:
```suggestion
Chakra
```
I think Chakra, or some other actual engine, makes more sense here. Unless it is intentional that we use something that isn't an engine.

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md [hunk 4]
```diff
@@ -17,375 +17,374 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+Starts the server
 
 ---
 
-Placeholder distractor 3
+Mounts middleware functions
```
Comment (lasjorg) [inline curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:186]: This distractor is the same as the answer.

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md [hunk 5]
```diff
@@ -17,375 +17,374 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+Building web applications and APIs
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is a major feature of Deno compared to Node.js?
```
Comment (lasjorg) [inline curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:130]: Not sure if this is relevant to a Node/Express quiz.

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md [hunk 6]
```diff
@@ -17,375 +17,374 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+Built-in TypeScript support
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is Bun primarily designed for?
```
Comment (lasjorg) [inline curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:152]: Same as with Deno.

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md [hunk 7]
```diff
@@ -17,375 +17,374 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+DELETE
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following is a popular ORM for Node.js?
```
Comment (lasjorg) [inline curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:262]: Technically, Mongoose is an ODM.

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md [hunk 8]
```diff
@@ -17,375 +17,374 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+npm run
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which JavaScript runtime is designed to be lightweight and optimized for performance?
```
Comment (lasjorg) [inline curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:349]: Again, I'm not sure different Runtimes are really relevant to this quiz. It is about Node and Express.

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md [hunk 9]
```diff
@@ -435,21 +434,21 @@ Placeholder answer
 
 #### --text--
 
-Placeholder question
+What type of SQL query is typically used to retrieve data from a database?
```
Comment (lasjorg) [inline curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:437]: As already said, it isn't about DBs.

**Review by lasjorg — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md (hunk 6 above)
Comment (ilenia-magoni) [reply curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:152]: Deno and Bun are within the topics listed in the issue, so I guess they are fine?

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (ilenia-magoni) [reply curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:240]: there will be probably, you can start fixing with this feedback

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md [hunk 10]
```diff
@@ -17,375 +17,374 @@ Answer all of the questions below correctly to pass the quiz.
...
 #### --distractors--
 
-Placeholder distractor 1
+req.body
 
 ---
 
-Placeholder distractor 2
-
+req.data
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:314]:
```suggestion
`req.data`

```

let's not remove empty lines

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md (hunk 6 above)
Comment (lasjorg) [reply curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:152]: I guess so. Just seems a little odd to me if the quiz is about Node and Express.

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md (hunk 3 above)
Comment (nitesh2920) [reply curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:76]: Yeah, I got your point. I will make the required changes

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md [hunk 11]
```diff
@@ -390,65 +389,65 @@ multer
 
 #### --text--
 
-Placeholder question
+Which of the following is true about browsers as JavaScript runtimes?
```
Comment (lasjorg) [inline curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:392]: I know the first point in the issue says

> JavaScript runtimes and engines (V8, node, deno, bun, browsers)

But that is super broad, questions about the browser seems completely unrelated to Node and Express.

I would much prefer a Node or Express question here. How about asking how you start an express server.

---

Which of the following methods are used to start an express server.

*distractors

`app.use`

`app.render`

`app.path`

*answer

`app.listen`

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md [hunk 12]
```diff
@@ -17,439 +17,437 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+express.bodyParser()
 
 #### --answer--
 
-Placeholder answer
+express.json()
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:102]:
```suggestion
`express.json()`
```

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md [hunk 13]
```diff
@@ -17,439 +17,437 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+3000
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the purpose of the cors package in an Express application?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:218]:
```suggestion
What is the purpose of the `cors` package in an Express application?
```

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md [hunk 14]
```diff
@@ -17,439 +17,437 @@ Answer all of the questions below correctly to pass the quiz.
...
-Placeholder answer
+To enable Cross-Origin Resource Sharing
 
 ### --question--
 
 #### --text--
 
-Placeholder question
-
+Which `npm` package is used to connect a Node.js application to a MySQL database?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:240]:
```suggestion
Which `npm` package is used to connect a Node.js application to a MySQL database?

```

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md [hunk 15]
```diff
@@ -17,439 +17,437 @@ Answer all of the questions below correctly to pass the quiz.
...
 #### --distractors--
 
-Placeholder distractor 1
+mongoose
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:243]:
```suggestion
`mongoose`
```

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md [hunk 16]
```diff
@@ -17,439 +17,437 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+sqlite
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:247]:
```suggestion
`sqlite`
```

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md [hunk 17]
```diff
@@ -17,439 +17,437 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+pg
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:251]:
```suggestion
`pg`
```

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md [hunk 18]
```diff
@@ -17,439 +17,437 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --answer--
 
-Placeholder answer
+mysql
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:255]:
```suggestion
`mysql`
```

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md [hunk 19]
```diff
@@ -17,439 +17,437 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+Mongoose
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the purpose of the dotenv package in a Node.js application?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:283]:
```suggestion
What is the purpose of the `dotenv` package in a Node.js application?
```

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md [hunk 20]
```diff
@@ -17,439 +17,437 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+To compile TypeScript
 
 #### --answer--
 
-Placeholder answer
+To load environment variables from a .env file
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:299]:
```suggestion
To load environment variables from a `.env` file
```

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md [hunk 21]
```diff
@@ -17,439 +17,437 @@ Answer all of the questions below correctly to pass the quiz.
...
 #### --distractors--
 
-Placeholder distractor 1
+`req.body`
 
 ---
 
-Placeholder distractor 2
-
+`req.data`
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:313]:
```suggestion
`req.data`

```

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md [hunk 22]
```diff
@@ -17,439 +17,437 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+`req.query`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which command is used to run scripts defined in the package.json file?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:326]:
```suggestion
Which command is used to run scripts defined in the `package.json` file?
```

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md [hunk 23]
```diff
@@ -17,439 +17,437 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --text--
 
-Placeholder question
+What middleware is commonly used for handling file uploads in Express?
 
 #### --distractors--
 
-Placeholder distractor 1
+express-fileupload
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:374]:
```suggestion
`express-fileupload`
```

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md [hunk 24]
```diff
@@ -17,439 +17,437 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+body-parser
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:378]:
```suggestion
`body-parser`
```

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md [hunk 25]
```diff
@@ -17,439 +17,437 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+compression
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:382]:
```suggestion
`compression`
```

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md [hunk 26]
```diff
@@ -17,439 +17,437 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --answer--
 
-Placeholder answer
+multer
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:386]:
```suggestion
`multer`
```

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md [hunk 27]
```diff
@@ -17,439 +17,437 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+They have a global 'window' object.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the purpose of the package-lock.json file?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:414]:
```suggestion
What is the purpose of the `package-lock.json` file?
```

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md [hunk 28]
```diff
@@ -17,439 +17,437 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --distractors--
 
-Placeholder distractor 1
+To manually specify package versions.
 
 ---
 
-Placeholder distractor 2
+To lock the npm version used in the project.
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:422]:
```suggestion
To lock the `npm` version used in the project.
```

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md [hunk 29]
```diff
@@ -17,439 +17,437 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+To store npm authentication tokens.
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:426]:
```suggestion
To store `npm` authentication tokens.
```

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md [hunk 30]
```diff
@@ -17,439 +17,437 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+To ensure consistent installs across environments.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+In a Node.js application, how do you create a connection to a MySQL database using the mysql package?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:436]:
```suggestion
In a Node.js application, how do you create a connection to a MySQL database using the `mysql` package?
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

**Review by lasjorg — COMMENTED** [review]
I'm OK with approving this if we want to keep the runtime questions.

But I would ask that you make the punctuation consistent. Anything where the distractor/answer is not a code block, but a word or a sentence should end with punctuation.

**Comment by nitesh2920** [comment]
> I'm OK with approving this if we want to keep the runtime questions.
> 
> But I would ask that you make the punctuation consistent. Anything where the distractor/answer is not a code block, but a word

> I'm OK with approving this if we want to keep the runtime questions.
> 
> But I would ask that you make the punctuation consistent. Anything where the distractor/answer is not a code block, but a word or a sentence should end with punctuation.

I have changed that browser runtime question.

**Comment by lasjorg** [comment]
> I have changed that browser runtime question.

I was talking about the Deno/Bun questions.

Anyway, the punctuations should be fixed.

**Comment by nitesh2920** [comment]
Ohh I got it , and I'm working on punctuation.

**Review by lasjorg — APPROVED** [review]

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md (hunk 21 above)
Comment (naomi-lgbt) [reply curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:313]: Nor this.

**Comment by lasjorg** [comment]
@nitesh2920 It is the missing new line in case you are wondering about the two outstanding change requests.

**Comment by nitesh2920** [comment]
> @nitesh2920 It is the missing new line in case you are wondering about the two outstanding change requests.

Okay, now I understand. Thank you, @lasjorg

File: curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md [hunk 31]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+`multer`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following methods are used to start an `express` server.?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-backend-javascript/66f1b11026394bcdb7c42473.md:394]:
```suggestion
Which of the following methods are used to start an `express` server?
```

**Review by naomi-lgbt — APPROVED** [review]
